### PR TITLE
refactor: use `throwOnError()`

### PR DESCRIPTION
**Pull request by iuioiua**
This reduces the need to throw errors using the returned error object explicitly.

### CODE REVIEW

File: tools/generate-embeddings.ts [hunk 1]
```diff
@@ -387,31 +377,24 @@ async function generateEmbeddings() {
...
+        await supabaseClient
           .from("dfods_page_section")
           .delete()
-          .filter("page_id", "eq", existingPage.id);
-
-        if (deletePageSectionError) {
-          throw deletePageSectionError;
-        }
+          .filter("page_id", "eq", existingPage.id)
+          .throwOnError;
```
Comment (thorwebdev) [inline tools/generate-embeddings.ts]:
```suggestion
          .throwOnError();
```

**Review by thorwebdev — APPROVED** [review]
I think this is more subjective preference. I personally prefer the previous version. But happy to let it go if other's prefer using `throwOnError`. In this case it probably makes sense ;)

**Comment by iuioiua** [comment]
Fair enough. My reasoning is that it reduces LOC without diluting what's going on thanks to the method's name.

**Comment by thorwebdev** [comment]
Yes, I agree that in this case it makes it more readable 👍

**Review by thorwebdev — APPROVED** [review]
LGTM 👍

**Comment by iuioiua** [comment]
CI issues will be fixed once #11 is merged.

**Pull request merged**